From aef385d44c19fbaef9473ca2f2fc9d3d23bc6ab5 Mon Sep 17 00:00:00 2001
From: Sofi <sofi+git@mailbox.org>
Date: Mon, 13 Feb 2023 14:44:15 +0100
Subject: [PATCH] use new ChangePassword interface

---
 LDAP-Auth/LDAPAuthenticationProviderPlugin.cs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs b/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs
index 517b85d..4a6c288 100644
--- a/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs
+++ b/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs
@@ -193,11 +193,12 @@ namespace Jellyfin.Plugin.LDAP_Auth
         /// Changes the users password (Requires privileged bind user).
         /// </summary>
         /// <param name="user">The user who's password will be changed.</param>
+        /// <param name="oldPassword">The previous password set.</param>
         /// <param name="newPassword">The new password to set.</param>
         /// <returns>Completed Task notification.</returns>
         /// <exception cref="NotImplementedException">Thrown if AllowPassChange set to false.</exception>
         /// <exception cref="InvalidOperationException">Thrown if LdapPasswordAttribute field is null or empty.</exception>
-        public Task ChangePassword(User user, string newPassword)
+        public Task ChangePassword(User user, string oldPassword, string newPassword)
         {
             if (!LdapPlugin.Instance.Configuration.AllowPassChange)
             {
-- 
2.39.1

