From 060e7e7496dac661c2beb4619d56d2ec2a7239e3 Mon Sep 17 00:00:00 2001
From: Sofi <sofi+git@mailbox.org>
Date: Fri, 24 Mar 2023 09:48:38 +0100
Subject: [PATCH] wip: allow `_file` extention on config keys

---
 canaille/__init__.py | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/canaille/__init__.py b/canaille/__init__.py
index fac778a..777e933 100644
--- a/canaille/__init__.py
+++ b/canaille/__init__.py
@@ -1,3 +1,4 @@
+from collections.abc import Mapping
 import datetime
 import os
 from logging.config import dictConfig
@@ -15,6 +16,18 @@ from .i18n import setup_i18n
 from .ldap_backend.backend import init_backend
 from .oidc.oauth import setup_oauth
 
+def parse_file_keys(config):
+    new_config = {}
+    for key, value in config.items():
+        if isinstance(value, Mapping):
+            new_config[key] = parse_file_keys(value)
+        elif isinstance(key, str) and key.endswith("_FILE") and isinstance(value, str):
+            with open(value, "r") as f:
+                value = f.read().rstrip("\n")
+            new_config[key[:-5]] = value
+        else:
+           new_config[key] = value
+    return new_config
 
 def setup_config(app, config=None, validate=True):
     import canaille.configuration
@@ -27,9 +40,9 @@ def setup_config(app, config=None, validate=True):
         }
     )
     if config:
-        app.config.from_mapping(config)
+        app.config.from_mapping(parse_file_keys(config))
     elif "CONFIG" in os.environ:
-        app.config.from_mapping(toml.load(os.environ.get("CONFIG")))
+        app.config.from_mapping(parse_file_keys(toml.load(os.environ["CONFIG"])))
     else:
         raise Exception(
             "No configuration file found. "
-- 
2.39.1

