From ba95cc13f10662a4154eaa2ca5080b468cde7d3b Mon Sep 17 00:00:00 2001
From: Sofi <sofi+git@mailbox.org>
Date: Thu, 25 May 2023 10:17:09 +0200
Subject: [PATCH] Use new changePassword interface

---
 LDAP-Auth/LDAPAuthenticationProviderPlugin.cs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs b/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs
index ce49b63..1c14a00 100644
--- a/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs
+++ b/LDAP-Auth/LDAPAuthenticationProviderPlugin.cs
@@ -194,11 +194,12 @@ namespace Jellyfin.Plugin.LDAP_Auth
         /// Changes the users password (Requires privileged bind user).
         /// </summary>
         /// <param name="user">The user who's password will be changed.</param>
+        /// <param name="oldPassword">The previous password set.</param>
         /// <param name="newPassword">The new password to set.</param>
         /// <returns>Completed Task notification.</returns>
         /// <exception cref="NotImplementedException">Thrown if AllowPassChange set to false.</exception>
         /// <exception cref="InvalidOperationException">Thrown if LdapPasswordAttribute field is null or empty.</exception>
-        public Task ChangePassword(User user, string newPassword)
+        public Task ChangePassword(User user, string  oldPassword, string newPassword)
         {
             if (!LdapPlugin.Instance.Configuration.AllowPassChange)
             {
-- 
2.40.1

